drivers/include/onewire.h

```diff
@@ -0,0 +1,246 @@
+ * therefore the ongoing data transfer will be invalid.
+ *
+ * To achieve reliable data transfers you have two choices:
+ * 1. Run the driver in a high priority thread that will not be interrupted.
+ *    Furthermore it needs to run in an environment, where at most very short
+ *    interrupts may be triggered.
+ * 2. Run the driver in safe mode by selecting the `onewire_safe` module. In
+ *    this mode all data transfers are run in a critical section by disabling
+ *    ALL interrupts for that period of time. Precisely, the interrupts will be
+ *    disabled for one block of 960us and 1-to-N blocks of 70us for each
+ *    reset->read/write sequence.
+ *
+ * @warning     Using the drivers safe mode (`onewire_safe`) will impact the
+ *              real-time capabilities of the overall system as interrupts will
+ *              be disabled for periods of time (blocks of 960us and 70us)
+ *
+ * @{
+ * @file
+ * @brief       1-Wire driver interface
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef ONEWIRE_H
+#define ONEWIRE_H
+
+#include <stdint.h>
+
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Length of 1-Wire ROM code in bytes
+ */
+#define ONEWIRE_ROM_LEN         (8U)
+
+/**
+ * @brief   String length of a hex encoded 1-Wire ROM code (including \0)
+ */
+#define ONEWIRE_ROM_STR_LEN     (17U)
+
+/**
+ * @brief   Custom value used to start a device search
+ */
+#define ONEWIRE_SEARCH_FIRST    (0)
+
+/**
+ * @brief   Return codes used by the 1-Wire driver
+ */
+enum {
+    ONEWIRE_OK          =  0,   /**< everything fine */
+    ONEWIRE_NODEV       = -1,   /**< no device connected to the bus */
+    ONEWIRE_ERR_PINCFG  = -2,   /**< unable to initialize pin as open-drain */
+    ONEWIRE_ERR_ROMSTR  = -3,   /**< invalid ROM string given */
+};
+
+/**
+ * @brief   1-Wire ROM commands
+ */
+enum {
+    ONEWIRE_ROM_SEARCH  = 0xf0, /**< search for devices */
+    ONEWIRE_ROM_READ    = 0x33, /**< read ROM code in single device config */
+    ONEWIRE_ROM_MATCH   = 0x55, /**< address a specific slave */
+    ONEWIRE_ROM_SKIP    = 0xcc, /**< broadcast commands to all connected devs */
+    ONEWIRE_ROM_ALARM   = 0xec, /**< search for devices with active alarm */
+};
+
+/**
+ * @brief   1-Wire configuration parameters
+ */
+typedef struct {
+    gpio_t pin;                 /**< GPIO pin the bus is connected to */
+    gpio_mode_t pin_mode;       /**< GPIO pin output mode */
+} onewire_params_t;
+
+/**
+ * @brief   1-Wire bus device descriptor
+ */
+typedef struct {
+    gpio_t pin;                 /**< GPIO pin */
+    gpio_mode_t omode;          /**< GPIO pin output mode */
+    gpio_mode_t imode;          /**< GPIO pin input mode, deducted from omode */
+} onewire_t;
+
+/**
+ * @brief   1-Wire ROM code (device address) representation
+ */
+typedef struct {
+    uint8_t u8[ONEWIRE_ROM_LEN];    /**< byte-wise access to address bytes */
+} onewire_rom_t;
```

Comment:
To me, `onewire_addr_t` would be easier to read. I assume most people would prefer the "normal" terminology over Dallas specific terminology.

Reply:
Personally I prefer to stick with the official terminology. Even if we don't like it, it seems the inventors of this tech like ROM better than ADDR. Why? No clue...